sdb/commands/linux/dmesg.py

```diff
@@ -0,0 +1,52 @@
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+from typing import Iterable
+
+import drgn
+import sdb
+
+
+class DMesg(sdb.Command):
+    # pylint: disable=too-few-public-methods
+
+    names = ["dmesg"]
+
+    def call(self, objs: Iterable[drgn.Object]) -> Iterable[drgn.Object]:
+        log_idx = self.prog["log_first_idx"]
+        log_seq = self.prog["clear_seq"]
+        log_end = self.prog["log_next_seq"]
+        log_buf = self.prog["log_buf"]
+
+        while log_seq < log_end:
+            entry = drgn.cast('struct printk_log *', log_buf + log_idx)
+
+            secs = int(entry.ts_nsec.value_() / 1000000000)
+            usecs = int((entry.ts_nsec.value_() % 1000000000) / 1000)
+
+            message = drgn.cast('char *', entry) + entry.type_.type.size
+            text = message.string_().decode('utf-8', 'ignore')
+
+            print("[{:5d}.{:06d}]: {:s}".format(secs, usecs, text))
+
+            if entry.len == 0:
+                log_idx = 0
+            else:
+                log_idx += entry.len
+            log_seq += 1
+        return []
```

Comment:
Since this doesn't return any objects, wouldn't be be better served as a `sdb.PrettyPrinter` object?

Reply:
This feels similar in nature to `zfs_dbgmsg`.. for that command, I chose to implement it as a locator and a pretty printer..

i.e. that command could be used to either return all debug messages (and can be piped), or it can be used to print all of the messages

Would it make sense to do something similar here?

Reply:
Rewrote this as a pretty-printer and locator, and retested.